Client/Frontend/Browser/BrowserViewController/BrowserViewController+WKNavigationDelegate.swift

```diff
@@ -258,6 +258,37 @@ extension BrowserViewController: WKNavigationDelegate {
                 self.tabManager.selectedTab?.alertShownCount = 0
                 self.tabManager.selectedTab?.blockAllAlerts = false
             }
+            
+            // Prevents synthetically activated links such as: CVE-2017-7089
+            if ["data", "blob", "file"].contains(url.scheme) {
+                if let clickType = navigationAction.value(forKey: "syntheticClickType") as? Int {
+                    //A click is synthetic if its value is 0 (aka WKSyntheticClickTypeNoTap).
+                    /*switch (syntheticClickType) {
+                    case WebKit::WebMouseEvent::OneFingerTap:
+                        return WKSyntheticClickTypeOneFingerTap;
+                    case WebKit::WebMouseEvent::TwoFingerTap:
+                        return WKSyntheticClickTypeTwoFingerTap;
+                    }*/
+                    if clickType == 0 {
+                        decisionHandler(.cancel)
+                        return
+                    }
+                }
+                
+                //Fallback.. Asks the user whether or not the url should be opened.. but only for `data`
+                if navigationAction.navigationType == .linkActivated && url.scheme == "data" {
+                    handleExternalURL(url) { didOpenURL in
```

Comment:
`data` is not an external URL. This code essentially disables `data urls` (which is tracked as a different issue). I tried this code snippet using this URL: `https://jumde.github.io/test/data-url.hml` and clicking `hello`, the data URL cannot be opened.